src/select.rs

```diff
@@ -121,6 +121,11 @@ impl<T: Display> Select<T> {
                     Key::Char('/') if self.filterable => self.handle_start_filtering(),
                     Key::Escape => self.handle_stop_filtering(false),
                     Key::Enter => {
+                        // if the cursor is on an empty option, don't return it
+                        if self.visible_options().get(self.cursor).is_none() {
```

Comment:
Not sure if there is a better option to handle this but it seems once search term is confirmed with `Enter` there is no way back to search term entry currently...